### DISCUSSION TITLE
Build for release takes a lot of time.

### DISCUSSION THREAD

**Discussion by saumitra91** [umijs/umi#9334] [Q&A]
Hi Umi!

So when I try to build my application for deployment using `cross-env UMI_ENV=staging umi build`, it takes about 9-15 minutes to build, which is a lot of build time.

Specifically, the build gets stuck on `92% chunk asset optimization` and causes the system to slow down, also takes up a lot of memory of RAM and swap.

Could you please suggest any solution for this?
I'd appreciate it.

**Comment by fz6m** [comment]
Please provide more information to help solve the problem  ( umi version / config / mini reproduce ... )

**Reply by saumitra91** [reply]
Hi Yingci,
This is for umi `3.4.25`.
The config is default, with some additional webpack plugins.

Also, I was wondering how I could swap `babel-loader` with `esbuild-loader`.
Could you let me know? 
I'd appreciate it.
